spring-integration-amqp/src/main/java/org/springframework/integration/amqp/dsl/AbstractMessageListenerContainerSpec.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.amqp.dsl;
+
+import java.util.concurrent.Executor;
+
+import org.aopalliance.aop.Advice;
+
+import org.springframework.amqp.core.AcknowledgeMode;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.util.ErrorHandler;
+
+/**
+ * Base class for container specs.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public abstract class AbstractMessageListenerContainerSpec<S extends AbstractMessageListenerContainerSpec<S, C>,
+		C extends AbstractMessageListenerContainer> {
+
+	private final C listenerContainer;
```

Comment:
Here is a polishing we discussed: https://github.com/artembilan/spring-integration/commit/38d84c355d1f8c720f332909b6d2fde25a42cafc

Yes, they can be registered as beans:
```
@Bean
MessageChannelSpec myChannel() {
     return MessageChannels.direct();
}
The most popular coding.

With the `FactoryBean` on board we can inject them regular way:
```
@Bean
IntegrationFlow myFlow(MessageChannel myChannel)
```
See `IntegrationFlowBeanPostProcessor.processIntegrationComponentSpec()` for more info.